### ISSUE TITLE
Too much recursion

### ISSUE THREAD

**Issue by vasikgreif** [formvalidation/support#73]
Hi, I cannot get rid of the `too much recursion` error. My code is now:
`
jQuery(document).ready(function ($) {

    $("[name='checkout']").on('init.field.fv', function(e, data) {
        var field  = data.field,        // Get the field name
            $field = data.element,      // Get the field element
            bv     = data.bv;           // BootstrapValidator instance

        // Create a span element to show valid message
        // and place it right before the field
        var $span = $('<small/>')
            .addClass('help-block validMessage alert alert-success')
            .attr('data-field', field)
            .insertAfter($field)
            .hide();

        // Retrieve the valid message via getOptions()
        var message = bv.getOptions(field).validMessage;
        if (message) {
            $span.html(message);
        }
    })
        .formValidation({
            framework: 'bootstrap',
        message: Ajax.required_field,
        fields: {
            order_domain_name: {
                excluded: false,
                container: '#order-domain-name-error',
                validMessage: 'Výborně, tato kombinace jméno a domény je dostupná',
                validators: {
                    notEmpty: {},
                    regexp: {
                        regexp: /^[a-z0-9]+$/i,
                        message: '<span class="alert alert-error">Název domény smí obsahovat jen malá písmena a čísla.</span>'
                    },
                    remote: {
                        message: '<span class="alert alert-error">' + Ajax.domain_already_registered + '</span>',
                        url: Ajax.ajaxurl,
                        type: "post",
                        data: function (validator) {
                            return {
                                domain: $('#order_domain').val(),
                                site: $("#order_domain_name").val(),
                                action: 'check_site_availability'
                            };
                        }
                    }
                }
            }
        }
    }).on('error.field.fv', function (e, data) {
            var field  = data.field,        // Get the field name
                $field = data.element;      // Get the field element

            // Show the valid message element
            $field.next('.validMessage[data-field="' + field + '"]').hide();
            data.bv.disableSubmitButtons(false);

        })
        .on('success.field.fv', function (e, data) {
        var field  = data.field,        // Get the field name
            $field = data.element;      // Get the field element

        // Show the valid message element
        $field.next('.validMessage[data-field="' + field + '"]').show();
        data.bv.disableSubmitButtons(false);
        });
});
`

Any ideas what's wrong?

**Comment by nghuuphuoc** [comment]
@vasikgreif 
Please give me your HTML form.
It might be caused by [form structure](http://formvalidation.io/getting-started/#writing-form).

**Comment by vasikgreif** [comment]
This is my form:
    <form action="http://demosite.com/objednavka/" class="checkout woocommerce-checkout row" id="checkout" method="post" name="checkout">
        <div class="span7" id="customer_details">
            <div>
                <h4 class="lined-heading"><span>Fakturační adresa</span></h4>
    
                <p class="form-row form-row-wide address-field update_totals_on_change validate-required" id="billing_country_field"><label class="">Země</label><strong>Česká
                        republika</strong><input class="country_to_state" id="billing_country" name="billing_country" type="hidden" value="CZ"></p>
    
                <p class="form-row form-row-first validate-required" id="billing_first_name_field"><label class="" for="billing_first_name">Křestní jméno <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="billing_first_name" name="billing_first_name" placeholder="" type="text" value="Vaclav"></p>
    
                <p class="form-row form-row-last validate-required" id="billing_last_name_field"><label class="" for="billing_last_name">Příjmení <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="billing_last_name" name="billing_last_name" placeholder="" type="text" value="Greif"></p>
    
                <div class="clear"></div>
    
                <p class="form-row form-row-wide" id="billing_company_field"><label class="" for="billing_company">Název firmy</label><input class="input-text" id="billing_company" name=
                    "billing_company" placeholder="" type="text" value=""></p>
    
                <p class="form-row form-row-wide address-field validate-required" id="billing_address_1_field"><label class="" for="billing_address_1">Adresa <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="billing_address_1" name="billing_address_1" placeholder="Ulice" type="text" value="Bezručova 178"></p>
    
                <p class="form-row form-row-wide address-field validate-required" id="billing_city_field"><label class="" for="billing_city">Město <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="billing_city" name="billing_city" placeholder="Město" type="text" value="Stod"></p>
    
                <p class="form-row form-row-first address-field validate-state" id="billing_state_field" style="display: none"><label class="" for="billing_state">Stát</label><input class="hidden"
                                                                                                                                                                                      id="billing_state" name="billing_state" placeholder="Stát" type="hidden" value=""></p>
    
                <p class="form-row form-row-last address-field validate-required validate-postcode" id="billing_postcode_field"><label class="" for="billing_postcode">PSČ / Zip <abbr class="required"
                                                                                                                                                                                       title="vyžadováno">*</abbr></label><input class="input-text" id="billing_postcode" name="billing_postcode" placeholder="PSČ / Zip" type="text" value="33301"></p>
    
                <div class="clear"></div>
    
                <p class="form-row form-row-first validate-required validate-email" id="billing_email_field"><label class="" for="billing_email">E-mailová adresa <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="billing_email" name="billing_email" placeholder="" type="text" value="vaclav.greif@gmail.com"></p>
    
                <p class="form-row form-row-last validate-required validate-phone" id="billing_phone_field"><label class="" for="billing_phone">Telefon <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="billing_phone" name="billing_phone" placeholder="" type="text" value="607963703"></p>
    
                <div class="clear"></div>
            </div>
    
            <div>
                <p class="form-row my-field-class form-row-wide validate-required" id="order_site_id_field"><label class="" for="order_site_id">Web <abbr class="required" title=
                        "vyžadováno">*</abbr></label><select class="select" id="order_site_id" name="order_site_id">
                        <option selected='selected' value="">
                            Vyberte web
                        </option>
    
                        <option value="2">
                            http://test.demosite.com
                        </option>
    
                        <option value="39">
                            http://ytataa.demosite.com
                        </option>
    
                        <option value="40">
                            http://hjhh000.demosite.com
                        </option>
    
                        <option value="41">
                            http://wbdx.demosite.com
                        </option>
    
                        <option value="42">
                            http://jlvo.demosite.com
                        </option>
    
                        <option value="43">
                            http://xfho.demosite.com
                        </option>
    
                        <option value="54">
                            http://ztdp.demosite.com
                        </option>
    
                        <option value="58">
                            http://xogx.demosite.com
                        </option>
    
                        <option value="66">
                            http://wsam.demosite.com
                        </option>
    
                        <option value="68">
                            http://kyck.demosite.com
                        </option>
    
                        <option value="69">
                            http://orvh.demosite.com
                        </option>
    
                        <option value="61">
                            http://phmc.demosite.com
                        </option>
    
                        <option value="71">
                            http://jkoq.demosite.com
                        </option>
    
                        <option value="75">
                            http://wapp.demosite.com
                        </option>
    
                        <option value="74">
                            http://hidg.demosite.com
                        </option>
    
                        <option value="73">
                            http://inmq.demosite.com
                        </option>
    
                        <option value="77">
                            http://ewkt.demosite.com
                        </option>
    
                        <option value="78">
                            http://acql.demosite.com
                        </option>
    
                        <option value="79">
                            http://ymag.demosite.com
                        </option>
    
                        <option value="82">
                            http://tvsy.demosite.com
                        </option>
    
                        <option value="84">
                            http://mjjo.demosite.com
                        </option>
    
                        <option value="87">
                            http://cvho.demosite.com
                        </option>
    
                        <option value="88">
                            http://vgwy.demosite.com
                        </option>
    
                        <option value="90">
                            http://ragu.demosite.com
                        </option>
    
                        <option value="91">
                            http://rgvk.demosite.com
                        </option>
    
                        <option value="93">
                            http://hltm.demosite.com
                        </option>
    
                        <option value="94">
                            http://axnu.demosite.com
                        </option>
    
                        <option value="99">
                            http://xaie.demosite.com
                        </option>
    
                        <option value="100">
                            http://sdjn.demosite.com
                        </option>
    
                        <option value="101">
                            http://qzcc.demosite.com
                        </option>
    
                        <option value="102">
                            http://iwtr.demosite.com
                        </option>
                    </select></p>
    
                <p class="form-row my-field-class form-row-wide validate-required" id="order_domain_name_field"><label class="" for="order_domain_name">Jméno <abbr class="required" title=
                        "vyžadováno">*</abbr></label><input class="input-text" id="order_domain_name" name="order_domain_name" placeholder="např. novakovi. Povolena jsou poze malá písmena a čísla." type=
                    "text" value=""></p>
    
                <p class="form-row my-field-class form-row-wide validate-required" id="order_domain_field"><label class="" for="order_domain">Doména <abbr class="required" title=
                        "vyžadováno">*</abbr></label><select class="select" id="order_domain" name="order_domain">
                        <option selected='selected' value="">
                            Select domain, where your site will be published
                        </option>
    
                        <option value="demosite.com">
                            demosite.com
                        </option>
    
                        <option value="gettingmarried.cz">
                            gettingmarried.cz
                        </option>
    
                        <option value="rekneme-si-ano.cz">
                            rekneme-si-ano.cz
                        </option>
    
                        <option value="svatba2015.cz">
                            svatba2015.cz
                        </option>
                    </select></p><span class="help-block" id="order-domain-name-error"></span>
                <hr>
    
                <h3 id="select_own_domain">Objednat vlastní doménu (Volitelné)</h3>
    
                <p>Pro Váš web můžete objednat vlastní doménu, např. www.novakovi.cz. Pro objednání vlastní domény zadejte vybranou doménu, kterou byste chtěli registrovat, a klikněte na tlačítko
                    'Ověřit dostupnost'.</p>
    
                <div class="own-domain-wrapper">
                    <p class="form-row my-field-class form-row-wide" id="custom_domain_field"><label class="" for="custom_domain">Doména</label><input class="input-text" id="custom_domain" name=
                        "custom_domain" placeholder="novakovi.cz" type="text" value=""></p><a class="btn btn-primary btn-sm ladda-button" data-style="slide-right" href="#" id=
                    "check_domain_availability"><span class="ladda-label">Zkontrolovat dostupnost</span></a>
    
                    <h3>Cena za doménu:</h3>
    
                    <p class="domain_price">Pro zobrazení ceny zadejte název domény a klikněte na tlačítko 'Ověřit dostupnost'.</p><span class="domain_price_value"></span> <span class=
                                                                                                                                                                                  "btn btn-primary btn-lg ladda-button" data-style="slide-right" id="add_domain_product" style="display: none;"><span class="ladda-label">Přidat doménu do objednávky</span></span>
                </div>
    
                <h3>Další informace</h3>
            </div>
        </div>
    
        <div class="col-2 span5" id="review-order">
            <div class="woocommerce-checkout-review-order review-order-wrap" id="order_review">
                <h4 id="order_review_heading"><span>Your Order</span></h4>
    
                <div id="order_review">
                    <table class="shop_table">
                        <thead>
                        <tr>
                            <th class="product-name">Produkt</th>
    
                            <th class="product-total">Celkem</th>
                        </tr>
                        </thead>
    
                        <tbody>
                        <tr class="cart_item">
                            <td class="product-name">Web demosite.com - basic <strong class="product-quantity">&times; 1</strong></td>
    
                            <td class="product-total"><span class="amount">1 500,00&#75;&#269;</span></td>
                        </tr>
                        </tbody>
    
                        <tfoot>
                        <tr class="cart-subtotal">
                            <th>Mezisoučet</th>
    
                            <td><span class="amount">1 500,00&#75;&#269;</span></td>
                        </tr>
    
                        <tr class="order-total">
                            <th>Cena celkem</th>
    
                            <td><strong><span class="amount">1 500,00&#75;&#269;</span></strong></td>
                        </tr>
                        </tfoot>
                    </table>
    
                    <div id="payment">
                        <ul class="payment_methods methods">
                            <li class="payment_method_gopay">
                                <input checked='checked' class="input-radio" data-order_button_text="" id="payment_method_gopay" name="payment_method" type="radio" value="gopay"> <label for=
                                                                                                                                                                                          "payment_method_gopay">GoPay - platby online kartou atd. <img alt="GoPay - platby online kartou atd." src=
                                    "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/gopay-blue.png"></label>
    
                                <div class="payment_box payment_method_gopay">
                                    <p>Zaplaťte kreditní kartou, mPlatbou, platbou České spořitelny a dalšími online platebními metodami.</p>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="cz_cs_c" name="paymentchannel" type="radio" value="cz_cs_c"> <img src=
                                                                                                                                                                  "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/eu_gp_u.png"> <span>Platba kartou</span>
                                    </div>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="SUPERCASH" name="paymentchannel" type="radio" value="SUPERCASH"> <img src=
                                                                                                                                                                      "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/SUPERCASH.png"> <span>superCash</span>
                                    </div>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="cz_kb" name="paymentchannel" type="radio" value="cz_kb"> <img src=
                                                                                                                                                              "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/cz_kb.png"> <span>KB Mojeplatba</span>
                                    </div>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="cz_rb" name="paymentchannel" type="radio" value="cz_rb"> <img src=
                                                                                                                                                              "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/cz_rb.png"> <span>Raiffeisenbank - ePlatby</span>
                                    </div>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="cz_mb" name="paymentchannel" type="radio" value="cz_mb"> <img src=
                                                                                                                                                              "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/cz_mb.png"> <span>mBank - mPeníze</span>
                                    </div>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="cz_fb" name="paymentchannel" type="radio" value="cz_fb"> <img src=
                                                                                                                                                              "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/cz_fb.png"> <span>Fio Banka</span>
                                    </div>
    
                                    <div class="gopay_select">
                                        <input checked="checked" class="gopay_select_input" id="cz_csas" name="paymentchannel" type="radio" value="cz_csas"> <img src=
                                                                                                                                                                  "http://demosite.com/wp-content/plugins/woo-gopay/public/assets/images/cz_csas.png"> <span>Česká spořitelna</span>
                                    </div>
                                </div>
                            </li>
    
                            <li class="payment_method_bacs">
                                <input class="input-radio" data-order_button_text="" id="payment_method_bacs" name="payment_method" type="radio" value="bacs"> <label for=
                                                                                                                                                                      "payment_method_bacs">Bankovním převodem</label>
    
                                <div class="payment_box payment_method_bacs" style="display:none;">
                                    <p>Proveďte platbu bankovním převodem. Jako variabilní symbol uveďte číslo objednávky. Váš web bude dostupný na zvolené adrese po připsání platby na náš účet.</p>
                                </div>
                            </li>
                        </ul>
    
                        <div class="form-row place-order">
                            <noscript>Protože Váš prohlížeč nepodporuje JavaScript, nebo je zakázaný, prosím, ujistěte se kliknutím na tlačítko <em>Přepočítat</em> před odesláním objednávky. Pokud
                                tak neučiníte, poplatky se mohou lišit od těch uvedených výše.<br>
                                <input class="button alt" name="woocommerce_checkout_update_totals" type="submit" value="Přepočítat"></noscript> <input id="_wpnonce" name="_wpnonce" type="hidden" value=
                            "45ecae2104"><input name="_wp_http_referer" type="hidden" value="/objednavka/">
    
                            <p class="form-row terms"><label class="checkbox" for="terms">Přečetl jsem si <a href="http://demosite.com/podminky/" target="_blank">obchodní podmínky</a> a souhlasím s
                                    nimi</label> <input class="input-checkbox required" id="terms" name="terms" required="" type="checkbox"></p><input class="button alt" data-value="Objednat" id=
                            "place_order" name="woocommerce_checkout_place_order" type="submit" value="Objednat">
                        </div>
    
                        <div class="clear"></div>
                    </div>
                </div>
            </div>
        </div>
    </form>

**Comment by nghuuphuoc** [comment]
What's the CSS framework you're using for your form?
I guess it's not Bootstrap, but you called

```javascript
$('[name="checkout"]').formValidation({
    framework: 'bootstrap',
    ...
});
```

And then the [too much recursion](http://formvalidation.io/getting-started/#writing-form) issue occurs because your form doesn't follow Bootstrap form structure.

**Comment by vasikgreif** [comment]
The site runs on Bootstrap, and I'm using formvalidation on another forms on the site. 

This particular form is a WooCommerce order form and I don't have a way to modify the structure of the HTML. How can I get rid of the error? I tried to add `row: '.form-row'`, but with no luck...
